azurerm/internal/services/kusto/registration.go

```diff
@@ -19,9 +19,10 @@ func (r Registration) SupportedDataSources() map[string]*schema.Resource {
 // SupportedResources returns the supported Resources supported by this Service
 func (r Registration) SupportedResources() map[string]*schema.Resource {
 	return map[string]*schema.Resource{
-		"azurerm_kusto_cluster":                  resourceArmKustoCluster(),
-		"azurerm_kusto_database":                 resourceArmKustoDatabase(),
-		"azurerm_kusto_database_principal":       resourceArmKustoDatabasePrincipal(),
-		"azurerm_kusto_eventhub_data_connection": resourceArmKustoEventHubDataConnection(),
+		"azurerm_kusto_cluster":                   resourceArmKustoCluster(),
+		"azurerm_kusto_database":                  resourceArmKustoDatabase(),
+		"azurerm_kusto_database_principal":        resourceArmKustoDatabasePrincipal(),
+		"azurerm_kusto_eventhub_data_connection":  resourceArmKustoEventHubDataConnection(),
+		"azurerm_kusto_eventgrid_data_connection": resourceArmKustoEventGridDataConnection(),
```

Comment:
Hey @mbfrahry thanks for comments, I will fix the order.

Regards accept test, 
Eventgrid data connection cannot be created without target table. 
Kusto RM doesn't provide a way to create table ( as it not part of the control plan)
So, I can't add test that doesn't depend on external resource.

@mbfrahry what do you recommend?

Reply:
Hey @noam-shapira, unfortunately I don't think we'll be able to take this in without being able to run nightly acceptance tests on it. Is the issue that Terraform doesn't currently manage the resource that's missing or there isn't a way for Terraform to manage it?

Reply:
@mbfrahry , sorry for the delay.
Tables and Mapping are not part of the control plan, so its not managed by arm resources.
I plan to add direct REST call to the resource, in order to create the Table and Mapping.

What do you think?

Reply:
If a way was provided in Terraform to execute Kusto management calls (.create-merge table, . create table xxx ingestion xx, ...) that would be FANTASTIC.  It's a significant hole for me right now (in addition to these data connections).  The data connections need created, but they won't work at all unless the tables and mappings are already there as well.
I had resigned myself to having terraform shell out to something I wrote myself to execute kusto management calls but I really don't want to have to resort to that.

Reply:
To add to that and why Terraform here would be wonderful - because Kusto requires tables/mappings to be created up-front having it be part of Terraform is pretty important.  Kusto however doesn't have the ability for idempotent changes - there's no create if not exist kind of operation.  .create-merge table is ok for table creations, but .create table xxx ingestion xx mapping will fail hard if the mapping already exists.  deleting a mapping that doesn't exist is also a hard error.
For terraform to offer idempotent table/mapping creation operations would be wonderful.

Reply:
Makes totally sense to me. But i would prefer to use the azure-sdk-for-go in favor of own REST calls. There is already a ticket opened by @mbfrahry to request the Kusto data support within the SDK: https://github.com/Azure/azure-sdk-for-go/issues/7916. As soon as this is solved i could start with an implementation. Until then we could use the ticket https://github.com/terraform-providers/terraform-provider-azurerm/issues/6392 to define a design of required resources.

Reply:
What about using https://github.com/Azure/azure-kusto-go for table and mapping creation? But i don't know if it's stable enough to use it.

Reply:
Actually, as much as I love Go, it unfortunately means either separately building a Go binary as part of a completely independent pipeline and getting it to where terraform is running, or having the Go compiler so that a go run could be executed.  The same configuration Terraform was using to authenticate with Azure would also need to be used by that same Go code.  It would then have to be a shelled-out 'local' execution tied to a resource.  

I will likely use the azure-kusto-go library for other things, but for the initial 'bake' of an environment using it (or the python lib, etc.) would likely be met with a bit of resistance.